File: app/src/main/java/com/perflyst/twire/fragments/AppearanceSettingsFragment.java
package com.perflyst.twire.fragments;


import android.os.Bundle;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.appcompat.content.res.AppCompatResources;
import androidx.fragment.app.Fragment;

import com.afollestad.materialdialogs.MaterialDialog;
import com.perflyst.twire.R;
import com.perflyst.twire.service.DialogService;
import com.perflyst.twire.service.Settings;

public class AppearanceSettingsFragment extends Fragment {
    private TextView themeSummary, streamsStyleSummary, gameStyleSummary, followStyleSummary, streamSizeSummary, gameSizeSummary, streamerSizeSummary;
    private ImageView themeSummaryColor;

    public static AppearanceSettingsFragment newInstance() {
        return new AppearanceSettingsFragment();
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_appearance_settings, container, false);
        themeSummary = rootView.findViewById(R.id.appearance_theme_color_summary);
        themeSummaryColor = rootView.findViewById(R.id.appearance_theme_color);

        streamsStyleSummary = rootView.findViewById(R.id.appearance_streams_style_summary);
        gameStyleSummary = rootView.findViewById(R.id.appearance_game_style_summary);
        followStyleSummary = rootView.findViewById(R.id.appearance_streamer_style_summary);

        streamSizeSummary = rootView.findViewById(R.id.appearance_streams_size_summary);
        gameSizeSummary = rootView.findViewById(R.id.appearance_game_size_summary);
        streamerSizeSummary = rootView.findViewById(R.id.appearance_streamer_size_summary);

        initSummaries();
        initOnClicks(rootView);

        return rootView;
    }

    private void initOnClicks(View rootView) {
        final View themeColorWrapper = rootView.findViewById(R.id.appearance_theme_color_wrapper);
        themeColorWrapper.setOnClickListener(v -> onClickThemeColor());

        final View streamStyleWrapper = rootView.findViewById(R.id.appearance_streams_style_wrapper);
        streamStyleWrapper.setOnClickListener(view -> onClickStreamStyle());

        final View streamSizeWrapper = rootView.findViewById(R.id.appearance_streams_size_wrapper);
        streamSizeWrapper.setOnClickListener(v -> onClickStreamSize());

        final View gameStyleWrapper = rootView.findViewById(R.id.appearance_game_style_wrapper);
        gameStyleWrapper.setOnClickListener(view -> onClickGameStyle());

        final View gameSizeWrapper = rootView.findViewById(R.id.appearance_game_size_wrapper);
        gameSizeWrapper.setOnClickListener(v1 -> onClickGameSize());

        final View streamerStyleWrapper = rootView.findViewById(R.id.appearance_streamer_style_wrapper);
        streamerStyleWrapper.setOnClickListener(view -> onClickStreamerStyle());

        final View streamerSizeWrapper = rootView.findViewById(R.id.appearance_streamer_size_wrapper);
        streamerSizeWrapper.setOnClickListener(v -> onClickStreamerSize());
    }

    private void initSummaries() {
        // Theme Summary
        themeSummary.setText(Settings.getTheme().getNameRes());
        themeSummaryColor.setImageDrawable(AppCompatResources.getDrawable(requireContext(), Settings.getTheme().chooser));

        // Style Summary
        streamsStyleSummary.setText(Settings.getAppearanceStreamStyle());
        gameStyleSummary.setText(Settings.getAppearanceGameStyle());
        followStyleSummary.setText(Settings.getAppearanceChannelStyle());

        // Size Summary
        streamSizeSummary.setText(Settings.getAppearanceStreamSize());
        gameSizeSummary.setText(Settings.getAppearanceGameSize());
        streamerSizeSummary.setText(Settings.getAppearanceChannelSize());
    }

    private void onClickThemeColor() {
        MaterialDialog themeChooserDialog = DialogService.getThemeDialog(getActivity());
        themeChooserDialog.show();
    }

    private void onClickStreamStyle() {
        DialogService.getChooseStreamCardStyleDialog(getActivity(), (title, index, previewView) -> {
            View sharedPadding = previewView.findViewById(R.id.shared_padding);
            ImageView view1 = previewView.findViewById(R.id.image_stream_preview);
            view1.setImageResource(R.drawable.preview_stream);
            View streamTitle = previewView.findViewById(R.id.stream_title);
            View viewersAndGame = previewView.findViewById(R.id.stream_game_and_viewers);

            if (title.equals(getString(R.string.card_style_expanded))) {
                streamTitle.setVisibility(View.VISIBLE);
                viewersAndGame.setVisibility(View.VISIBLE);
                sharedPadding.setVisibility(View.VISIBLE);
            } else if (title.equals(getString(R.string.card_style_normal))) {
                streamTitle.setVisibility(View.GONE);
                viewersAndGame.setVisibility(View.VISIBLE);
                sharedPadding.setVisibility(View.VISIBLE);
            } else if (title.equals(getString(R.string.card_style_minimal))) {
                streamTitle.setVisibility(View.GONE);
                viewersAndGame.setVisibility(View.GONE);
                sharedPadding.setVisibility(View.GONE);
            }

            Settings.setAppearanceStreamStyle(title);
            initSummaries();

        }).show();
    }

    private void onClickGameStyle() {
        DialogService.getChooseGameCardStyleDialog(getActivity(), (title, index, previewView) -> {
            View sharedPadding = previewView.findViewById(R.id.shared_padding);
            ImageView view1 = previewView.findViewById(R.id.image_game_preview);
            view1.setImageResource(R.drawable.preview_game);

            FrameLayout.LayoutParams lp = new FrameLayout.LayoutParams(
                    (int) requireContext().getResources().getDimension(R.dimen.game_preview_width),
                    ViewGroup.LayoutParams.WRAP_CONTENT
            );
            lp.gravity = Gravity.CENTER;
            previewView.setLayoutParams(lp);

            View gameTitle = previewView.findViewById(R.id.game_card_title);
            /*
            TextView gameViewers = previewView.findViewById(R.id.game_viewers);
            if (gameViewers.getText().equals("")) {
                gameViewers.setText(R.string.preview_game_viewers);
            }
             */

            if (title.equals(getString(R.string.card_style_expanded))) {
                gameTitle.setVisibility(View.VISIBLE);
                //gameViewers.setVisibility(View.VISIBLE);
                sharedPadding.setVisibility(View.VISIBLE);
            } else if (title.equals(getString(R.string.card_style_normal))) {
                gameTitle.setVisibility(View.GONE);
                //gameViewers.setVisibility(View.VISIBLE);
                sharedPadding.setVisibility(View.VISIBLE);
            } else if (title.equals(getString(R.string.card_style_minimal))) {
                gameTitle.setVisibility(View.GONE);
                //gameViewers.setVisibility(View.GONE);
                sharedPadding.setVisibility(View.GONE);
            }

            Settings.setAppearanceGameStyle(title);
            initSummaries();
        }).show();
    }

    private void onClickStreamerStyle() {
        DialogService.getChooseStreamerCardStyleDialog(getActivity(), (title, index, previewView) -> {
            View nameView = previewView.findViewById(R.id.displayName);
            ImageView streamerLogo = previewView.findViewById(R.id.profileLogoImageView);
            streamerLogo.setImageResource(R.drawable.preview_streamer);

            FrameLayout.LayoutParams lp = new FrameLayout.LayoutParams(
                    (int) requireContext().getResources().getDimension(R.dimen.subscription_card_preview_width),
                    ViewGroup.LayoutParams.WRAP_CONTENT
            );
            lp.gravity = Gravity.CENTER;
            previewView.setLayoutParams(lp);

            if (title.equals(getString(R.string.card_style_normal))) {
                nameView.setVisibility(View.VISIBLE);
            } else if (title.equals(getString(R.string.card_style_minimal))) {
                nameView.setVisibility(View.GONE);
            }

            Settings.setAppearanceChannelStyle(title);
            initSummaries();
        }).show();
    }

    private void onClickStreamSize() {
        DialogService.getChooseCardSizeDialog(
                requireActivity(),
                R.string.appearance_streams_size_title,
                Settings.getAppearanceStreamSize(),
                (dialog, itemView, which, text) -> {
                    Settings.setAppearanceStreamSize(text.toString());
                    initSummaries();
                    return true;
                }
        ).show();
    }

    private void onClickStreamerSize() {
        DialogService.getChooseCardSizeDialog(
                requireActivity(),
                R.string.appearance_streamer_size_title,
                Settings.getAppearanceChannelSize(),
                (dialog, itemView, which, text) -> {
                    Settings.setAppearanceChannelSize(text.toString());
                    initSummaries();
                    return true;
                }
        ).show();
    }

    private void onClickGameSize() {
        DialogService.getChooseCardSizeDialog(
                requireActivity(),
                R.string.appearance_game_size_title,
                Settings.getAppearanceGameSize(),
                (dialog, itemView, which, text) -> {
                    Settings.setAppearanceGameSize(text.toString());
                    initSummaries();
                    return true;
                }
        ).show();
    }
}

File: app/src/main/java/com/perflyst/twire/fragments/AppearanceSettingsFragment.kt
package com.perflyst.twire.fragments

import android.os.Bundle
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.content.res.AppCompatResources
import androidx.fragment.app.Fragment
import com.afollestad.materialdialogs.MaterialDialog
import com.perflyst.twire.R
import com.perflyst.twire.service.DialogService
import com.perflyst.twire.service.Settings.appearanceChannelSize
import com.perflyst.twire.service.Settings.appearanceChannelStyle
import com.perflyst.twire.service.Settings.appearanceGameSize
import com.perflyst.twire.service.Settings.appearanceGameStyle
import com.perflyst.twire.service.Settings.appearanceStreamSize
import com.perflyst.twire.service.Settings.appearanceStreamStyle
import com.perflyst.twire.service.Settings.theme


class AppearanceSettingsFragment : Fragment() {
    private lateinit var themeSummary: TextView
    private lateinit var streamsStyleSummary: TextView
    private lateinit var gameStyleSummary: TextView
    private lateinit var followStyleSummary: TextView
    private lateinit var streamSizeSummary: TextView
    private lateinit var gameSizeSummary: TextView
    private lateinit var streamerSizeSummary: TextView
    private lateinit var themeSummaryColor: ImageView

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        val rootView = inflater.inflate(R.layout.fragment_appearance_settings, container, false)
        themeSummary = rootView.findViewById(R.id.appearance_theme_color_summary)
        themeSummaryColor = rootView.findViewById(R.id.appearance_theme_color)

        streamsStyleSummary = rootView.findViewById(R.id.appearance_streams_style_summary)
        gameStyleSummary = rootView.findViewById(R.id.appearance_game_style_summary)
        followStyleSummary = rootView.findViewById(R.id.appearance_streamer_style_summary)

        streamSizeSummary = rootView.findViewById(R.id.appearance_streams_size_summary)
        gameSizeSummary = rootView.findViewById(R.id.appearance_game_size_summary)
        streamerSizeSummary = rootView.findViewById(R.id.appearance_streamer_size_summary)

        initSummaries()
        initOnClicks(rootView)

        return rootView
    }

    private fun initOnClicks(rootView: View) {
        val themeColorWrapper = rootView.findViewById<View>(R.id.appearance_theme_color_wrapper)
        themeColorWrapper.setOnClickListener { v: View? -> onClickThemeColor() }

        val streamStyleWrapper = rootView.findViewById<View>(R.id.appearance_streams_style_wrapper)
        streamStyleWrapper.setOnClickListener { view: View? -> onClickStreamStyle() }

        val streamSizeWrapper = rootView.findViewById<View>(R.id.appearance_streams_size_wrapper)
        streamSizeWrapper.setOnClickListener { v: View? -> onClickStreamSize() }

        val gameStyleWrapper = rootView.findViewById<View>(R.id.appearance_game_style_wrapper)
        gameStyleWrapper.setOnClickListener { view: View? -> onClickGameStyle() }

        val gameSizeWrapper = rootView.findViewById<View>(R.id.appearance_game_size_wrapper)
        gameSizeWrapper.setOnClickListener { v1: View? -> onClickGameSize() }

        val streamerStyleWrapper =
            rootView.findViewById<View>(R.id.appearance_streamer_style_wrapper)
        streamerStyleWrapper.setOnClickListener { view: View? -> onClickStreamerStyle() }

        val streamerSizeWrapper = rootView.findViewById<View>(R.id.appearance_streamer_size_wrapper)
        streamerSizeWrapper.setOnClickListener { v: View? -> onClickStreamerSize() }
    }

    private fun initSummaries() {
        // Theme Summary
        themeSummary.setText(theme.nameRes)
        themeSummaryColor.setImageDrawable(
            AppCompatResources.getDrawable(
                requireContext(),
                theme.chooser
            )
        )

        // Style Summary
        streamsStyleSummary.text = appearanceStreamStyle
        gameStyleSummary.text = appearanceGameStyle
        followStyleSummary.text = appearanceChannelStyle

        // Size Summary
        streamSizeSummary.text = appearanceStreamSize
        gameSizeSummary.text = appearanceGameSize
        streamerSizeSummary.text = appearanceChannelSize
    }

    private fun onClickThemeColor() {
        val themeChooserDialog = DialogService.getThemeDialog(requireActivity())
        themeChooserDialog.show()
    }

    private fun onClickStreamStyle() {
        DialogService.getChooseStreamCardStyleDialog(
            requireActivity()
        ) { title: String?, index: Int, previewView: View? ->
            val sharedPadding = previewView!!.findViewById<View>(R.id.shared_padding)
            val view1 = previewView.findViewById<ImageView>(R.id.image_stream_preview)
            view1.setImageResource(R.drawable.preview_stream)
            val streamTitle = previewView.findViewById<View>(R.id.stream_title)
            val viewersAndGame = previewView.findViewById<View>(R.id.stream_game_and_viewers)

            when (title) {
                getString(R.string.card_style_expanded) -> {
                    streamTitle.visibility = View.VISIBLE
                    viewersAndGame.visibility = View.VISIBLE
                    sharedPadding.visibility = View.VISIBLE
                }

                getString(R.string.card_style_normal) -> {
                    streamTitle.visibility = View.GONE
                    viewersAndGame.visibility = View.VISIBLE
                    sharedPadding.visibility = View.VISIBLE
                }

                getString(R.string.card_style_minimal) -> {
                    streamTitle.visibility = View.GONE
                    viewersAndGame.visibility = View.GONE
                    sharedPadding.visibility = View.GONE
                }
            }

            appearanceStreamStyle = title!!
            initSummaries()
        }.show()
    }

    private fun onClickGameStyle() {
        DialogService.getChooseGameCardStyleDialog(
            requireActivity()
        ) { title: String?, index: Int, previewView: View? ->
            val sharedPadding = previewView!!.findViewById<View>(R.id.shared_padding)
            val view1 = previewView.findViewById<ImageView>(R.id.image_game_preview)
            view1.setImageResource(R.drawable.preview_game)

            val lp = FrameLayout.LayoutParams(
                requireContext().resources.getDimension(R.dimen.game_preview_width)
                    .toInt(),
                ViewGroup.LayoutParams.WRAP_CONTENT
            )
            lp.gravity = Gravity.CENTER
            previewView.setLayoutParams(lp)

            val gameTitle = previewView.findViewById<View>(R.id.game_card_title)

            /*
        TextView gameViewers = previewView.findViewById(R.id.game_viewers);
        if (gameViewers.getText().equals("")) {
            gameViewers.setText(R.string.preview_game_viewers);
        }
         */
            when (title) {
                getString(R.string.card_style_expanded) -> {
                    gameTitle.visibility = View.VISIBLE
                    //gameViewers.setVisibility(View.VISIBLE);
                    sharedPadding.visibility = View.VISIBLE
                }

                getString(R.string.card_style_normal) -> {
                    gameTitle.visibility = View.GONE
                    //gameViewers.setVisibility(View.VISIBLE);
                    sharedPadding.visibility = View.VISIBLE
                }

                getString(R.string.card_style_minimal) -> {
                    gameTitle.visibility = View.GONE
                    //gameViewers.setVisibility(View.GONE);
                    sharedPadding.visibility = View.GONE
                }
            }

            appearanceGameStyle = title!!
            initSummaries()
        }.show()
    }

    private fun onClickStreamerStyle() {
        DialogService.getChooseStreamerCardStyleDialog(
            requireActivity()
        ) { title: String?, index: Int, previewView: View? ->
            val nameView = previewView!!.findViewById<View>(R.id.displayName)
            val streamerLogo = previewView.findViewById<ImageView>(R.id.profileLogoImageView)
            streamerLogo.setImageResource(R.drawable.preview_streamer)

            val lp = FrameLayout.LayoutParams(
                requireContext().resources
                    .getDimension(R.dimen.subscription_card_preview_width).toInt(),
                ViewGroup.LayoutParams.WRAP_CONTENT
            )
            lp.gravity = Gravity.CENTER
            previewView.setLayoutParams(lp)

            if (title == getString(R.string.card_style_normal)) {
                nameView.visibility = View.VISIBLE
            } else if (title == getString(R.string.card_style_minimal)) {
                nameView.visibility = View.GONE
            }

            appearanceChannelStyle = title!!
            initSummaries()
        }.show()
    }

    private fun onClickStreamSize() {
        DialogService.getChooseCardSizeDialog(
            requireActivity(),
            R.string.appearance_streams_size_title,
            appearanceStreamSize
        ) { dialog: MaterialDialog?, itemView: View?, which: Int, text: CharSequence? ->
            appearanceStreamSize = text.toString()
            initSummaries()
            true
        }.show()
    }

    private fun onClickStreamerSize() {
        DialogService.getChooseCardSizeDialog(
            requireActivity(),
            R.string.appearance_streamer_size_title,
            appearanceChannelSize
        ) { dialog: MaterialDialog?, itemView: View?, which: Int, text: CharSequence? ->
            appearanceChannelSize = text.toString()
            initSummaries()
            true
        }.show()
    }

    private fun onClickGameSize() {
        DialogService.getChooseCardSizeDialog(
            requireActivity(),
            R.string.appearance_game_size_title,
            appearanceGameSize
        ) { dialog: MaterialDialog?, itemView: View?, which: Int, text: CharSequence? ->
            appearanceGameSize = text.toString()
            initSummaries()
            true
        }.show()
    }

    companion object {
        fun newInstance(): AppearanceSettingsFragment {
            return AppearanceSettingsFragment()
        }
    }
}
